### ISSUE TITLE
RightCurlyCheck: False Positive with SAME or ALONE_OR_SINGLELINE

### ISSUE THREAD

**Issue by sd1998** [checkstyle/checkstyle#7187]
https://checkstyle.org/config_blocks.html#RightCurly

`RightCulryCheck.java` gives false positives in case of `RightCurlyOption.SAME` and `RightCurlyOptioon.ALONE_OR_SINGLELINE` in case of tokens like `CLASS_DEF`, `METHOOD_DEF`, `ANNOOTATION_DEF` etc.

```
$ javac YOUR_FILE.java

$ cat test_config_same.xml

<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
            <module name="RightCurly">
             <property name="tokens" 
        value="LITERAL_TRY, LITERAL_CATCH, LITERAL_FINALLY, LITERAL_IF, 
                    LITERAL_ELSE, CLASS_DEF, METHOD_DEF, CTOR_DEF, 
                    LITERAL_FOR, LITERAL_WHILE, LITERAL_DO, 
                    STATIC_INIT, INSTANCE_INIT"/>
            <property name="option" value="same"/>
    </module>
    </module>
</module>

$ cat test_config_alone_or_single.xml

<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
            <module name="RightCurly">
             <property name="tokens" 
                    value="LITERAL_TRY, LITERAL_CATCH, LITERAL_FINALLY, LITERAL_IF, 
                               LITERAL_ELSE, CLASS_DEF, METHOD_DEF, CTOR_DEF, 
                               LITERAL_FOR, LITERAL_WHILE, LITERAL_DO, 
                               STATIC_INIT, INSTANCE_INIT"/>
            <property name="option" value="alone_or_singleline"/>
    </module>
    </module>
</module>

$ cat TestClass.java

public class TestClass {

	public TestClass() {}; //violation
	
	public void doSomething(){}; //violation
}; //violation

$ java -Duser.language=en -Duser.country=US -jar checkstyle-8.26-all.jar \
            -c test_config_same.xml TestClass.java

Starting audit...
[ERROR] /TestClass.java:3:29: '}' at column 22 should be alone on a line. [RightCurly]
[ERROR] /TestClass.java:5:35: '}' at column 28 should be alone on a line. [RightCurly]
[ERROR] /TestClass.java:6:1: '}' at column 1 should be alone on a line. [RightCurly]
Audit done.
Checkstyle ends with 3 errors.

$ java -Duser.language=en -Duser.country=US -jar checkstyle-8.26-all.jar \
     -c test_config_alone_or_single.xml TestClass.java

Starting audit...
[ERROR] /TestClass.java:3:29: '}' at column 22 should be alone on a line. [RightCurly]
[ERROR] /TestClass.java:5:35: '}' at column 28 should be alone on a line. [RightCurly]
[ERROR] /TestClass.java:6:1: '}' at column 1 should be alone on a line. [RightCurly]
Audit done.
Checkstyle ends with 3 errors.
```

As can seen above in case of `RightCurlyOption.SAME` and `RightCurlyOption.ALONE_OR_SINGLELINE` a violation is raised which should not be the case.

**Comment by rnveach** [comment]
@sd1998 Make sure your example actual works and is compilable.
`shouldStartLine` was removed in 8.17 and `Class` is not a valid Java syntax.

**Comment by sd1998** [comment]
@rnveach Thanks. `Class` was a typo.

**Comment by sd1998** [comment]
@rnveach I have a question this case of false positives also exists across `METHOD_DEF` and other DEFINITION_TOKENS as well. And the fix for all them is similar so should I fix them all in this issue itself.

**Comment by rnveach** [comment]
As long as it doesn't make the issue too complex to the users to understand and the PR really complex to review, I am fine with it. It also doesn't hurt to make each a separate issue so it is all straightforward and clear.

If the issue will be expanded, the title and description will have to be updated.

**Comment by romani** [comment]
@sd1998 , please update issue to strictly follow template, I need to make sure that you used our CLI to reproduce issue.

**Comment by sd1998** [comment]
@romani I have updated the issue description.

**Comment by romani** [comment]
Please convert images to text and please provide output as text for RightCurlyOptioon.ALONE_OR_SINGLELINE

**Comment by sd1998** [comment]
@romani I have updated the issue desription.

**Comment by romani** [comment]
Cases are almost same as at examples at https://checkstyle.org/property_types.html#rcurly , the only difference is that in examples blocks are not empty.
Issue is valid.

**Comment by romani** [comment]
Fix is merged
